### PR TITLE
English translations

### PR DESCRIPTION
**Pull request by mattias-p**
Here are English translations for newly introduced messages, and also a few drive-by tweaks of old ones. Since none of the other translation files refer to the new msgids yet, I took the liberty to update the msgids themselves. Having correct srings in the msgids makes it easier to make the other translations.

### CODE REVIEW

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 1]
```diff
@@ -112,45 +112,45 @@ sub metadata {
...
     return {
         ARE_AUTHORITATIVE    => "All these nameservers are confirmed to be authoritative : {nsset}.",
-        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in child are unique.",
-        CHILD_NS_SAME_IP     => "IP {address} in child refers to multiple nameservers ({nss}).",
-        DEL_DISTINCT_NS_IP   => "All the IP addresses used by the nameservers in parent are unique.",
-        DEL_NS_SAME_IP       => "IP {address} in parent refers to multiple nameservers ({nss}).",
+        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in the zone are unique.",
+        CHILD_NS_SAME_IP     => "IP {address} in the zone refers to multiple nameservers ({nss}).",
+        DEL_DISTINCT_NS_IP   => "All the IP addresses used by the nameservers in the delegation are unique.",
+        DEL_NS_SAME_IP       => "IP {address} in the delegation refers to multiple nameservers ({nss}).",
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: I think we should consider being more explicit, e.g.:

```
CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in the zone (the domain name tested) are unique.",
CHILD_NS_SAME_IP     => "IP {address} in the zone (the domain name tested) refers to multiple nameservers ({nss}).",
DEL_DISTINCT_NS_IP   => "All the IP addresses used by the nameservers in the delegation of the domain name tested are unique.",
DEL_NS_SAME_IP       => "IP {address} in the delegation of the domain name tested refers to multiple nameservers ({nss}).",
```

Is it a problem if the message is long? What is the limit? Could it be several sentences?

What do you think, @sandoche2k?

Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: We should make an effort to keep the messages short and concise but having a having a hard limit on message length doesn't make much sense.

Zone and delegation are very central concepts. About one quarter of the messages refer to them. Repeating the phrase "the domain tested" in all of those messages doesn't make much sense to me.

Comment (sandoche2k) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: I agree with @mattias-p. Short is better but with no restrictions. Long messages are never read :-)

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 2]
```diff
@@ -348,13 +348,13 @@ sub metadata {
 sub translation {
     return {
         ADDITIONAL_DNSKEY_SKIPPED => "No DNSKEYs found. Additional tests skipped.",
-        ALGORITHM_DELETE_DS       => "The DNSKEY with tag {keytag} uses Delete DS algorithm number "
+        ALGORITHM_DELETE_DS       => "The DNSKEY with tag {keytag} uses the Delete DS algorithm number "
```
Comment (sandoche2k) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: the => a

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 3]
```diff
@@ -112,45 +112,45 @@ sub metadata {
 sub translation {
     return {
         ARE_AUTHORITATIVE    => "All these nameservers are confirmed to be authoritative : {nsset}.",
-        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in child are unique.",
-        CHILD_NS_SAME_IP     => "IP {address} in child refers to multiple nameservers ({nss}).",
-        DEL_DISTINCT_NS_IP   => "All the IP addresses used by the nameservers in parent are unique.",
-        DEL_NS_SAME_IP       => "IP {address} in parent refers to multiple nameservers ({nss}).",
+        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in the zone are unique.",
```
Comment (sandoche2k) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: Child zone seems to be appropriate for me

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 1 above)
Comment (sandoche2k) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: Parent zone seems to be appropriate to me

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 2 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: Really? There's only one algorithm number that means Delete DS, so in my opinion "the" is better.

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 3 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: I like "the zone" and "the delegation" better. "The zone" is the one we're testing and "the delegation" is its delegation.
If we're using the "parent"/"child" terminology, it's clear that we're dealing with two zones, but it's (slightly) less obvious which ones.

Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: I'd like to point out that RFC 7199 was obsoleted by [RFC 8499](https://tools.ietf.org/html/rfc8499), but that doesn't invalidate the point you're making.

I agree that the use of "delegation" in my suggestion doesn't fit with the definitions in the RFCs. I'll revert to "child"/"parent".

**Comment by mattias-p** [comment]
@matsduf, @sandoche2k, @vlevigneron Please check again.

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 3 above)
Comment (matsduf) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: It is correct that [RFC 8499] defines "delegation" as "[t]he process by which a separate zone is created" but the RFC itself uses "delegation" to also mean the DNS records that creates the delegation, e.g. "(...) the same as the origin of the zone that contains the **delegation** to the name server (...)" (my emphasis).

My conclusion is that there is no problem to use "delegation" to refer the the DNS records that create the delegation. It is used in that sense over and over again by DNS experts.

[RFC 8499]: https://tools.ietf.org/html/rfc8499

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 4]
```diff
@@ -112,8 +112,8 @@ sub metadata {
 sub translation {
     return {
         ARE_AUTHORITATIVE    => "All these nameservers are confirmed to be authoritative : {nsset}.",
-        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in child are unique.",
-        CHILD_NS_SAME_IP     => "IP {address} in child refers to multiple nameservers ({nss}).",
+        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in the zone are unique.",
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: It is ambigues to refer to "the zone" when we have two zones. I suggest "the zone for the domain being tested". The argument is that it is too long, but complex things cannot always be explained in a short sentence.

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 5]
```diff
@@ -146,11 +146,11 @@ sub translation {
           . "Lower limit set to {minimum}.",
         NO_IPV4_NS_CHILD => "Child lists no nameserver that resolves to an IPv4 address. If any were present, "
           . "the minimum allowed would be {minimum}.",
-        NO_IPV4_NS_DEL => "Delegation lists no nameserver that resolves to an IPv4 address. If any were present, "
+        NO_IPV4_NS_DEL => "Parent lists no nameserver that resolves to an IPv4 address. If any were present, "
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: "Parent" --> "Parent zone" or "Parent zone (parent domain)"

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 6]
```diff
@@ -146,11 +146,11 @@ sub translation {
...
           . "the minimum allowed would be {minimum}.",
         NO_IPV6_NS_CHILD => "Child lists no nameserver that resolves to an IPv6 address. If any were present, "
           . "the minimum allowed would be {minimum}.",
-        NO_IPV6_NS_DEL => "Delegation lists no nameserver that resolves to an IPv6 address. If any were present, "
+        NO_IPV6_NS_DEL => "Parent lists no nameserver that resolves to an IPv6 address. If any were present, "
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: "Parent" --> "Parent zone" or "Parent zone (parent domain)"

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 4 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: I created issue zonemaster/zonemaster#760 to create a glossary for technical terms. I suggest that "the zone" is added as a term to the glossary with the definition "the zone for the domain being tested". To contrast this we could define "the parent zone" to mean "the parent zone for the domain being tested", or possibly "the parent zone of the zone for the domain being tested", if you prefer to be explicit.

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 5 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: Reverting to old wording.

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 6 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: Reverting to old wording.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 7]
```diff
@@ -405,7 +405,7 @@ sub translation {
         NO_KEYS_OR_NO_SIGS_OR_NO_SOA => "Cannot test SOA signatures, because we got {keys} DNSKEY records, "
           . "{sigs} RRSIG records and {soas} SOA records.",
         NO_NSEC3PARAM          => "{server} returned no NSEC3PARAM records.",
-        NO_RESPONSE_DNSKEY     => "Nameserver {ns}/{address} responded with no DNSKEY record(s).",
+        NO_RESPONSE_DNSKEY     => "Nameserver {ns}/{address} responded without DNSKEY record(s).",
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm]: "without DNSKEY" --> "without any DNSKEY"

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 3 above)
Comment (matsduf) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: "All the IP addresses used by the nameservers in the zone are unique." --> "The IP addresses of the nameservers listed in the zone of the tested domain are unique."

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 8]
```diff
@@ -112,8 +112,8 @@ sub metadata {
 sub translation {
     return {
         ARE_AUTHORITATIVE    => "All these nameservers are confirmed to be authoritative : {nsset}.",
-        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in child are unique.",
-        CHILD_NS_SAME_IP     => "IP {address} in child refers to multiple nameservers ({nss}).",
+        CHILD_DISTINCT_NS_IP => "All the IP addresses used by the nameservers in the zone are unique.",
+        CHILD_NS_SAME_IP     => "IP {address} in the zone refers to multiple nameservers ({nss}).",
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: "IP {address} in the zone refers to multiple nameservers ({nss})." --> "Of the nameservers listed in the zone of the tested domain there are multiple such ({nss}) having the IP address {address}."

**Comment by mattias-p** [comment]
@matsduf and @sandoche2k, please go through your review comment and resolve the conversations where appropriate.

**Comment by matsduf** [comment]
> @matsduf and @sandoche2k, please go through your review comment and resolve the conversations where appropriate.

Done.

**Comment by mattias-p** [comment]
@matsduf I didn't mean to make a re-request for your review. I mis-clicked.

**Comment by mattias-p** [comment]
I'll start over from scratch.

**Pull request closed without merging**